stability: panic in tracing code

relevant log lines (also note the 24 second Raft `handleRaftReady` time preceding the panic):

```
Table/51/1/6737663152663509586/"d4e6eb07-a66b-4c0a-89b8-acd41ef2b545"/3745): replica {3 3 3} too old, added to replica GC queue
W160822 05:24:40.408088 storage/store.go:2473  store=3:3 range=291 [/Table/51/1/8511432968882437492/"df6fee99-02a1-4c3b-bf12-1970a0da3ff4"/2562-/Table/51/1/8545829731955717928/"eb728016-bd8c-4ffa-b508-6a16d36b54fb"/3021): handle raft ready: 24.0s
W160822 05:24:40.408156 storage/store.go:2473  store=3:3: raft ready processing: 24.0s
I160822 05:24:40.424950 storage/store.go:2405  store=3:3 range=289 [/Table/51/1/3665892545080726987/"ce74d92c-c921-4388-853d-bed92058c88b"/1265-/Table/51/1/3699184130198378280/"74600e78-9fb4-4471-9215-3b44c0f34ceb"/831): replica {3 3 2} too old, added to replica GC queue
I160822 05:24:40.453056 storage/store.go:2405  store=3:3 range=208 [/Table/51/1/3050589115531691471/"1870e682-efe2-4ada-844d-a5be812a31e7"/120-/Table/51/1/3086603066031747482/"e7a6b562-557f-498f-9950-4f8789abd3a8"/679): replica {3 3 6} too old, added to replica GC queue
I160822 05:24:40.519204 storage/store.go:2405  store=3:3 range=282 [/Table/51/1/1986183688238978389/"91f4033f-30a0-43f1-86b0-546bbd28bbb8"/724-/Table/51/1/2021951297381760072/"fb488ffc-fdbc-4f4f-8eda-4896e462cfa8"/1697): replica {3 3 7} too old, added to replica GC queue
I160822 05:24:40.544900 storage/store.go:2405  store=3:3 range=29 [/Table/51/1/6595505570928908425/"1a9f803e-a44a-4bf5-bc8e-bfa29f228cd0"/723-/Table/51/1/6632279873513956123/"eb728016-bd8c-4ffa-b508-6a16d36b54fb"/2860): replica {3 3 6} too old, added to replica GC queue
panic: (*basictracer.errAssertionFailed) (0x14f2e60,0xc42bd632e0) [recovered]
	panic: (*basictracer.errAssertionFailed) (0x14f2e60,0xc42bd632e0)

goroutine 84 [running]:
panic(0x14f2e60, 0xc42bd632e0)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/cockroachdb/cockroach/util/stop.(*Stopper).Recover(0xc42046a510)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:176 +0x6e
panic(0x14f2e60, 0xc42bd632e0)
	/usr/local/go/src/runtime/panic.go:458 +0x243
github.com/opentracing/basictracer-go.(*spanImpl).maybeAssertSanityLocked(0xc42046a3f0)
	/go/src/github.com/opentracing/basictracer-go/debug.go:27 +0x331
github.com/opentracing/basictracer-go.(*spanImpl).Lock(0xc42046a3f0)
	/go/src/github.com/opentracing/basictracer-go/debug.go:19 +0x3f
github.com/opentracing/basictracer-go.(*spanImpl).Log(0xc42046a3f0, 0x0, 0x0, 0x0, 0xc44f2297a0, 0x54, 0x0, 0x0)
	/go/src/github.com/opentracing/basictracer-go/span.go:106 +0x83
github.com/opentracing/basictracer-go.(*spanImpl).LogEvent(0xc42046a3f0, 0xc44f2297a0, 0x54)
	/go/src/github.com/opentracing/basictracer-go/span.go:94 +0x9e
github.com/cockroachdb/cockroach/util/log.Trace(0x7f4a99a681f8, 0xc4203eee40, 0xc44f2297a0, 0x54)
	/go/src/github.com/cockroachdb/cockroach/util/log/trace.go:33 +0xc7
github.com/cockroachdb/cockroach/util/log.addStructured(0x7f4a99a681f8, 0xc4203eee40, 0xc400000001, 0x2, 0x16d367a, 0x21, 0xc43773beb8, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/util/log/structured.go:62 +0x10d
github.com/cockroachdb/cockroach/util/log.logDepth(0x7f4a99a681f8, 0xc4203eee40, 0x1, 0x7f4a00000001, 0x16d367a, 0x21, 0xc43773beb8, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/util/log/log.go:70 +0x8c
github.com/cockroachdb/cockroach/util/log.Warningf(0x7f4a99a681f8, 0xc4203eee40, 0x16d367a, 0x21, 0xc43773beb8, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/util/log/log.go:103 +0x7e
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func2()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:1039 +0x20a
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc42046a510, 0xc420777160)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x7d
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x66
```

full logs are in the artifacts for this test run:

https://teamcity.cockroachdb.com/viewLog.html?buildId=13469&buildTypeId=Cockroach_AllocatorTest3to510g&tab=buildLog#_focus=1707

--- a/storage/store.go
+++ b/storage/store.go
@@ -953,7 +953,7 @@ func (s *Store) Start(ctx context.Context, stopper *stop.Stopper) error {
 		// sentinel and first range metadata if we have a first range.
 		// This may wake up ranges and requires everything to be set up and
 		// running.
-		s.startGossip(ctx)
+		s.startGossip()
 
 		// Start the scanner. The construction here makes sure that the scanner
 		// only starts after Gossip has connected, and that it does not block Start
@@ -1001,11 +1001,12 @@ func (s *Store) WaitForInit() {
 // startGossip runs an infinite loop in a goroutine which regularly checks
 // whether the store has a first range or config replica and asks those ranges
 // to gossip accordingly.
-func (s *Store) startGossip(ctx context.Context) {
+func (s *Store) startGossip() {
 	// Periodic updates run in a goroutine and signal a WaitGroup upon completion
 	// of their first iteration.
 	s.initComplete.Add(2)
 	s.stopper.RunWorker(func() {
+		ctx := s.context(context.Background())
 		// Run the first time without waiting for the Ticker and signal the WaitGroup.
 		if err := s.maybeGossipFirstRange(); err != nil {
 			log.Warningf(ctx, "error gossiping first range data: %s", err)
@@ -1026,6 +1027,7 @@ func (s *Store) startGossip(ctx context.Context) {
 	})
 
 	s.stopper.RunWorker(func() {
+		ctx := s.context(context.Background())
 		if err := s.maybeGossipSystemConfig(); err != nil {
 			log.Warningf(ctx, "error gossiping system config: %s", err)
 		}

